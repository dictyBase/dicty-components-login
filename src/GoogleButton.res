external styleToString: ReactDOM.Style.t => string = "%identity"

module Styles = %makeStyles(
  theme => {
    button: ReactDOM.Style.make(
      ~paddingTop=`${theme.spacing(2)->string_of_int}px`,
      ~paddingBottom=`${theme.spacing(2)->string_of_int}px`,
      ~paddingLeft=`${theme.spacing(3)->string_of_int}px`,
      ~paddingRight=`${theme.spacing(3)->string_of_int}px`,
      ~textTransform=`none`,
      ~backgroundColor="#df4b37",
      (),
    )->ReactDOM.Style.unsafeAddProp(
      "&:hover",
      ReactDOM.Style.make(~backgroundColor="#a60e0e", ())->styleToString,
    ),
    subtitle: ReactDOM.Style.make(~width="100%", ~color="#fafafa", ()),
  }
)

@genType @react.component
let make = (~text: string, ~handleClick: ReactEvent.Mouse.t => unit) => {
  let classes = Styles.useStyles()

  <MaterialUi_Button
    variant=#Contained
    size=#Large
    fullWidth={true}
    startIcon={<GoogleIcon />}
    className={classes.button}
    onClick={handleClick}>
    <MaterialUi_Typography
      className={classes.subtitle} variant=#H6 align=#Center color=#TextSecondary>
      {text}
    </MaterialUi_Typography>
  </MaterialUi_Button>
}

let default = make
