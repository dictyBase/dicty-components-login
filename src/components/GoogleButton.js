// @flow
import React from "react"
import { Flex, Box } from "rebass"
import FontAwesomeIcon from "@fortawesome/react-fontawesome"
import faGoogle from "@fortawesome/fontawesome-free-brands/faGoogle"
import { FontAwesomeIconContainer, GoogleButtonStyle } from "../styles/Buttons"
/**
 * This is the Google social button. It has a FontAwesome icon and then some sign in text.
 */

const GoogleButton = () => {
  return (
    <Flex justify="center">
      <Box w={"90%"} mb={"5px"}>
        <GoogleButtonStyle size="large" variant="raised">
          <FontAwesomeIconContainer>
            <FontAwesomeIcon icon={faGoogle} />
          </FontAwesomeIconContainer>&nbsp; Sign in with Google
        </GoogleButtonStyle>
      </Box>
    </Flex>
  )
}

export default GoogleButton
