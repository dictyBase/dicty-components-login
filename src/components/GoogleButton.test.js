import React from "react"
import renderer from "react-test-renderer"
import GoogleButton from "./GoogleButton"
import "jest-styled-components"

test("matching a snapshot of GoogleButton", () => {
  const component = renderer.create(<GoogleButton />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
