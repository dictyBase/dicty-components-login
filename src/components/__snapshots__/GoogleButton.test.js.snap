// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of GoogleButton 1`] = `
.c1 {
  box-sizing: border-box;
}

.c2 {
  box-sizing: border-box;
  width: 90%;
  margin-bottom: 5px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  border-right: 1px solid black;
  line-height: 45px;
  width: 45px;
  font-size: 1.8em;
}

.c3 {
  display: inline-block;
  padding: 6px 12px;
  margin-bottom: 0;
  font-size: 14px;
  font-weight: normal;
  line-height: 1.42857143;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-image: none;
  border: 1px solid transparent;
  border-radius: 4px;
  position: relative;
  padding-left: 21px;
  text-align: left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-top: 5px;
  color: #fff;
  border-color: rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 10px 16px;
  font-size: 18px;
  line-height: 1.3333333;
  border-radius: 6px;
  background-color: #dd4b39;
}

.c3.block {
  display: block;
  width: 100%;
}

.c3.large {
  padding: 10px 16px;
  font-size: 18px;
  line-height: 1.3333333;
  border-radius: 6px;
}

.c3.xs {
  padding: 1px 5px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 3px;
  width: 20%;
}

.c3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:focus,
.c3:hover,
.c3:active {
  color: #fff;
  background-color: #c23321;
  border-color: rgba(0,0,0,0.2);
}

<div
  className="c0 c1"
  is={null}
>
  <div
    className="c2"
    is={null}
  >
    <a
      className="c3"
    >
      <i
        className="c4"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-google fa-w-16 "
          data-icon="google"
          data-prefix="fab"
          role="img"
          style={Object {}}
          viewBox="0 0 488 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </i>
      Â  Sign in with Google
    </a>
  </div>
</div>
`;
